File: pinyougou-cart-interface/src/main/java/cn/pinyougou/cart/service/CartService.java
package cn.pinyougou.cart.service;


import cn.pinyougou.pojogroup.Cart;

import java.util.List;

/**
 * 购物车服务接口
 */
public interface CartService {

    /**
     * 添加商品到购物车
     * @param cartList 购物车列表 （先拿到购物车列表操作后返回）
     * @param itemId  商品id
     * @param num   数量
     * @return
     */
    public List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);

    /**
     * 从redis中查询购物车
     * @param username
     * @return
     */
    public List<Cart> findCartListFromRedis(String username);

    /**
     * 将购物车保存到redis中
     * @param username
     * @param cartList
     */
    public void saveCartListToRedis(String username,List<Cart> cartList);

    /**
     * 合并购物车
     * @param cartList1
     * @param cartList2
     * @return
     */
    public List<Cart> mregeCartList(List<Cart> cartList1,List<Cart> cartList2);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService', function ($http) {
    //购物车列表
    this.findCartList = function () {
        return $http.get('cart/findCartList.do');
    }
    //数量的加减 添加商品到购物车
    this.addGoodsToCartList = function (itemId, num) {
        return $http.get('cart/addGoodsToCartList.do?itemId=' + itemId + '&num=' + num);
    }

    //求合计
    this.sum = function (cartList) {
        //合计实体
        var totalValue={totalNum:0,totalMoney:0};
        //遍历购物车获取每个购物车对象
        for(var i=0;i<cartList.length;i++){
            var cart=cartList[i];//购物车对象
            //遍历购物车对象得到购物车明细列表
            for(var j=0;j<cart.orderItemList.length;j++){
                 var orderItem=cart.orderItemList[j];//购物车明细
                totalValue.totalNum +=orderItem.num;//累加数量
                totalValue.totalMoney += orderItem.totalFee;//累加金额
            }
        }
     return totalValue;
    }
});